src/transport.js

```diff
@@ -53,11 +54,18 @@ module.exports = function (swarm) {
       // specific for the transport we are using
       const proxyConn = new Connection()
 
-      next(multiaddrs.shift())
+      each(multiaddrs, next)
 
-      // TODO improve in the future to make all the dials in paralell
+      var success = false
       function next (multiaddr) {
+        if (success) {
+          return;
+        }
         const conn = t.dial(multiaddr, () => {
+          if (success) {
+            return;
```

Comment:
This is leaving open sockets behind. We need to check: `If I have successfully opened a socket before, then I need to close this one.`